src/test/java/uk/gov/ons/census/casesvc/service/NewAddressReportedServiceTest.java

```diff
@@ -0,0 +1,233 @@
+    OffsetDateTime expectedDateTime = OffsetDateTime.now();
+
+    // When
+    underTest.processNewAddress(newAddressEvent, expectedDateTime);
+
+    // Then
+    Case expectedCase = getExpectedCase(collectionCase);
+
+    ArgumentCaptor<Case> caseArgumentCaptor = ArgumentCaptor.forClass(Case.class);
+    verify(caseService).saveNewCaseAndStampCaseRef(caseArgumentCaptor.capture());
+    Case actualCase = caseArgumentCaptor.getValue();
+    assertThat(actualCase).isEqualToComparingFieldByFieldRecursively(expectedCase);
+
+    verify(caseService).emitCaseCreatedEvent(casetoEmit);
+
+    verify(eventLogger)
+        .logCaseEvent(
+            eq(casetoEmit),
+            eq(eventDTO.getDateTime()),
+            eq("New Address reported"),
+            eq(EventType.NEW_ADDRESS_REPORTED),
+            eq(eventDTO),
+            eq(JsonHelper.convertObjectToJson(newAddressEvent.getPayload())),
+            eq(expectedDateTime));
+  }
+
+  @Test
+  public void testMinimalGoodNewAddressDoesNotFailValidation() {
+    ResponseManagementEvent responseManagementEvent = getMinimalValidNewAddress();
+    underTest.processNewAddress(responseManagementEvent, OffsetDateTime.now());
+  }
+
+  @Test(expected = RuntimeException.class)
+  public void testMissingUUID() {
+    ResponseManagementEvent newAddressEvent = getMinimalValidNewAddress();
+    newAddressEvent.getPayload().getNewAddress().getCollectionCase().setId("Xdd344234");
+
+    try {
+      underTest.processNewAddress(newAddressEvent, OffsetDateTime.now());
+    } catch (RuntimeException e) {
+      assertThat(e.getMessage())
+          .isEqualTo("Expected NewAddress CollectionCase Id to be a valid UUID, got: Xdd344234");
+      throw e;
+    }
+  }
+
+  @Test(expected = RuntimeException.class)
+  public void testBadCasetype() {
+    ResponseManagementEvent newAddressEvent = getMinimalValidNewAddress();
+    newAddressEvent.getPayload().getNewAddress().getCollectionCase().setCaseType("BadCaseType");
+
+    try {
+      underTest.processNewAddress(newAddressEvent, OffsetDateTime.now());
+    } catch (RuntimeException e) {
+      assertThat(e.getMessage())
+          .isEqualTo("Unexpected newAddress CollectionCase caseType: BadCaseType");
+      throw e;
+    }
+  }
+
+  @Test(expected = RuntimeException.class)
+  public void testBadAddressLevel() {
+    ResponseManagementEvent newAddressEvent = getMinimalValidNewAddress();
+    newAddressEvent
+        .getPayload()
+        .getNewAddress()
+        .getCollectionCase()
+        .getAddress()
+        .setAddressLevel("X");
+
+    try {
+      underTest.processNewAddress(newAddressEvent, OffsetDateTime.now());
+    } catch (RuntimeException e) {
+      assertThat(e.getMessage())
+          .isEqualTo(
+              "Unexpected a valid address level in newAddress CollectionCase Address, received: X");
+      throw e;
+    }
+  }
+
+  @Test(expected = RuntimeException.class)
+  public void testBadRegion() {
```

Comment:
Not "bad". **Missing**

The requirements say "check the minimum mandatory fields are present"